### PR TITLE
system variable: add `tidb_restricted_read_only`

### PR DESCRIPTION
**Pull request by en-jin19**
### What is changed, added or deleted? (Required)
Added the description of following configurations:
- `RESTRICTED_REPLICA_WRITER_ADMIN` in the "privilege-management.md"
- `tidb_restricted_read_only` in the "system-variables.md"

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6716
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by ichn-hu — APPROVED** [review]
LGTM

File: system-variables.md [hunk 1]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
 
+### tidb_restricted_read_only <span class="version-mark">New in v5.2.0</span>
+
+- Scope: GLOBAL
+- Default value: `0`
+- Optional values: `0`, `1`
+- This variable controls the read-only status of the entire cluster. If the variable is enabled (the value is `1`), all TiDB servers in the entire cluster turn on the read-only mode. In this case, TiDB only execute the statements that do not modify data, such as `SELECT`, `USE`, `SHOW`. For others, such as `INSERT`, `UPDATE`, TiDB rejects to execute those statements in the read-only mode.
+- The read-only mode enabled by this variable only ensures that the entire cluster finally goes into the read-only status. If you have changed the value of this variable, but the changed status is not updated to other TiDB servers, the un-updated TiDB is still **not** in the read-only mode.
+- When enabling this variable, the executing SQL statements are not affected. TiDB only checks the read-only status for the SQL statements that are to be executed.
+- When enabling this variable, the uncommitted transactions can have the following results:
```
Comment (en-jin19) [inline system-variables.md:765]:
```suggestion
- During enabling this variable, the executing SQL statements are not affected. TiDB only checks the read-only status for the SQL statements to be executed.
- During enabling this variable, the uncommitted transactions can have the following results:
```

File: system-variables.md [hunk 2]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+    - If there is an uncommitted read-only transaction, the transaction can be committed normally.
+    - If the uncommitted transaction is not a read-only transaction, the SQL statements executing write operations in that transaction are rejected.
+    - If the uncommitted read-only transaction has modified data, its commitment is also rejected.
```
Comment (en-jin19) [inline system-variables.md:768]:
```suggestion
    - If the uncommitted transaction is not a read-only transaction, the SQL statements executing write operations in the transaction are rejected.
    - If the uncommitted read-only transaction has modified data, the commitment of the transaction is rejected.
```

File: system-variables.md [hunk 3]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+- After the cluster enables read-only mode, all users (including the users with the `SUPER` privilege) cannot execute the SQL statements that might write data unless the user is explicitly granted the `RESTRICTED_REPLICA_WRITER_ADMIN` privileges.
```
Comment (en-jin19) [inline system-variables.md:769]:
```suggestion
- After the read-only mode is enabled, unless the user is explicitly granted the `RESTRICTED_REPLICA_WRITER_ADMIN` privileges, all users (including the users with the `SUPER` privilege) cannot execute the SQL statements that might write data.
```

File: system-variables.md [hunk 4]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+- The users with `RESTRICTED_VARIABLES_ADMIN` or `SUPER` privileges can modify this variable. If you enabled [Security Enhanced Mode](#tidb_enable_enhanced_security), only the users with the `RESTRICTED_VARIABLES_ADMIN` privilege can modify this variable.
```
Comment (en-jin19) [inline system-variables.md:770]:
```suggestion
- The users with `RESTRICTED_VARIABLES_ADMIN` or `SUPER` privileges can modify this variable. If the [Security Enhanced Mode](#tidb_enable_enhanced_security) is enabled, only the users with the `RESTRICTED_VARIABLES_ADMIN` privilege can modify this variable.
```

File: system-variables.md [hunk 5]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
     - `RESTRICTED_VARIABLES_ADMIN`: The ability to see and set sensitive variables in `SHOW [GLOBAL] VARIABLES` and `SET`.
     - `RESTRICTED_USER_ADMIN`: The ability to prevent other users from making changes or dropping a user account.
 
+### tidb_restricted_read_only <span class="version-mark">New in v5.2.0</span>
+
+- Scope: GLOBAL
+- Default value: `0`
+- Optional values: `0`, `1`
+- This variable controls the read-only status of the entire cluster. If the variable is enabled (the value is `1`), all TiDB servers in the entire cluster turn on the read-only mode. In this case, TiDB only execute the statements that do not modify data, such as `SELECT`, `USE`, `SHOW`. For others, such as `INSERT`, `UPDATE`, TiDB rejects to execute those statements in the read-only mode.
```
Comment (en-jin19) [inline system-variables.md:762]:
```suggestion
- This variable controls the read-only status of the entire cluster. If the variable is enabled (the value is `1`), all TiDB servers in the entire cluster turn on the read-only mode. In this case, TiDB only executes the statements that do not modify data, such as `SELECT`, `USE`, `SHOW`. For others, such as `INSERT` and `UPDATE`, TiDB rejects to execute those statements in the read-only mode.
```

File: system-variables.md [hunk 6]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+- The read-only mode enabled by this variable only ensures that the entire cluster finally goes into the read-only status. If you have changed the value of this variable, but the changed status is not updated to other TiDB servers, the un-updated TiDB is still **not** in the read-only mode.
+- During enabling this variable, the executing SQL statements are not affected. TiDB only checks the read-only status for the SQL statements to be executed.
```
Comment (en-jin19) [inline system-variables.md:764]:
```suggestion
- If you enabled the read-only mode by this variable, this only ensures that the entire cluster finally goes into the read-only status. Suppose that you have changed the value of this variable, but the changed status is not updated to other TiDB servers. In this situation, the un-updated TiDB is still **not** in the read-only mode.
- While enabling this variable, the executing SQL statements are not affected. TiDB only checks the read-only status for the SQL statements **to be** executed.
```

File: system-variables.md [hunk 7]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+- During enabling this variable, the uncommitted transactions can have the following results:
+    - If there is an uncommitted read-only transaction, the transaction can be committed normally.
+    - If the uncommitted transaction is not a read-only transaction, the SQL statements executing write operations in the transaction are rejected.
+    - If the uncommitted read-only transaction has modified data, the commitment of the transaction is rejected.
```
Comment (en-jin19) [inline system-variables.md:768]:
```suggestion
- While enabling this variable, the uncommitted transactions have the following results:
    - For uncommitted read-only transactions, the transactions can be committed normally.
    - For uncommitted transactions that are not read-only transactions, SQL statements executing write operations in these transactions are rejected.
    - For uncommitted read-only transactions with modified data, commitments of these transactions are rejected.
```

File: system-variables.md [hunk 8]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+- After the read-only mode is enabled, unless the user is explicitly granted the `RESTRICTED_REPLICA_WRITER_ADMIN` privileges, all users (including the users with the `SUPER` privilege) cannot execute the SQL statements that might write data.
+- The users with `RESTRICTED_VARIABLES_ADMIN` or `SUPER` privileges can modify this variable. However, if the [Security Enhanced Mode](#tidb_enable_enhanced_security) is enabled, only the users with the `RESTRICTED_VARIABLES_ADMIN` privilege can modify this variable.
```
Comment (en-jin19) [inline system-variables.md:770]:
```suggestion
- After the read-only mode is enabled, all users (including the users with the `SUPER` privilege) cannot execute the SQL statements that might write data unless the user is explicitly granted the `RESTRICTED_REPLICA_WRITER_ADMIN` privileges.
- The users with `RESTRICTED_VARIABLES_ADMIN` or `SUPER` privileges can modify this variable. However, if the [Security Enhanced Mode](#tidb_enable_enhanced_security) is enabled, only the users with the `RESTRICTED_VARIABLES_ADMIN` privilege can modify this variable.
```

File: privilege-management.md [hunk 9]
```diff
@@ -195,6 +195,7 @@ Dynamic privileges include:
 * `ROLE_ADMIN`
 * `CONNECTION_ADMIN`
 * `SYSTEM_VARIABLES_ADMIN`
+* `RESTRICTED_REPLICA_WRITER_ADMIN` allows privilege owners to perform write or update operations without being affected when the TiDB cluster is enabled the read-only mode. For details, see [`tidb_restricted_read_only`](/system-variables.md#tidb_restricted_read_only-new-in-v520).
```
Comment (TomShawn) [inline privilege-management.md:198]:
```suggestion
* `RESTRICTED_REPLICA_WRITER_ADMIN` allows privilege owners to perform write or update operations without being affected when the read-only mode is enabled in the TiDB cluster. For details, see [`tidb_restricted_read_only`](/system-variables.md#tidb_restricted_read_only-new-in-v520).
```

File: system-variables.md [hunk 10]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
     - `RESTRICTED_VARIABLES_ADMIN`: The ability to see and set sensitive variables in `SHOW [GLOBAL] VARIABLES` and `SET`.
     - `RESTRICTED_USER_ADMIN`: The ability to prevent other users from making changes or dropping a user account.
 
+### tidb_restricted_read_only <span class="version-mark">New in v5.2.0</span>
```
Comment (TomShawn) [inline system-variables.md:757]: 建议按字母顺序放在下方

File: system-variables.md [hunk 11]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+
+- Scope: GLOBAL
+- Default value: `0`
+- Optional values: `0`, `1`
```
Comment (TomShawn) [inline system-variables.md:761]:
```suggestion
- Value optipns: `0`, `1`
```

File: system-variables.md [hunk 12]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+- This variable controls the read-only status of the entire cluster. If the variable is enabled (which means the value is `1`), all TiDB servers in the entire cluster turn on the read-only mode. In this case, TiDB only executes the statements that do not modify data, such as `SELECT`, `USE`, `SHOW`. For others, such as `INSERT` and `UPDATE`, TiDB rejects to execute those statements in the read-only mode.
```
Comment (TomShawn) [inline system-variables.md:762]: "reject to do" is incorrect.
```suggestion
- This variable controls the read-only status of the entire cluster. If the variable is enabled (which means that the value is `1`), all TiDB servers in the entire cluster are in the read-only mode. In this case, TiDB only executes the statements that do not modify data, such as `SELECT`, `USE`, and `SHOW`. For other statements such as `INSERT` and `UPDATE`, TiDB rejects executing those statements in the read-only mode.
```

File: system-variables.md [hunk 13]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+- If you enabled the read-only mode by this variable, this only ensures that the entire cluster finally goes into the read-only status. Suppose that you have changed the value of this variable, but the changed status is not updated to other TiDB servers. In this situation, the un-updated TiDB is still **not** in the read-only mode.
+- While enabling this variable, the executing SQL statements are not affected. TiDB only checks the read-only status for the SQL statements **to be** executed.
```
Comment (TomShawn) [inline system-variables.md:764]:
```suggestion
- While this variable is enabled, the SQL statements being executed are not affected. TiDB only performs the read-only check for the SQL statements **to be** executed.
```

File: system-variables.md [hunk 14]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
     - `RESTRICTED_VARIABLES_ADMIN`: The ability to see and set sensitive variables in `SHOW [GLOBAL] VARIABLES` and `SET`.
     - `RESTRICTED_USER_ADMIN`: The ability to prevent other users from making changes or dropping a user account.
 
+### tidb_restricted_read_only <span class="version-mark">New in v5.2.0</span>
+
+- Scope: GLOBAL
+- Default value: `0`
+- Optional values: `0`, `1`
+- This variable controls the read-only status of the entire cluster. If the variable is enabled (which means the value is `1`), all TiDB servers in the entire cluster turn on the read-only mode. In this case, TiDB only executes the statements that do not modify data, such as `SELECT`, `USE`, `SHOW`. For others, such as `INSERT` and `UPDATE`, TiDB rejects to execute those statements in the read-only mode.
+- If you enabled the read-only mode by this variable, this only ensures that the entire cluster finally goes into the read-only status. Suppose that you have changed the value of this variable, but the changed status is not updated to other TiDB servers. In this situation, the un-updated TiDB is still **not** in the read-only mode.
```
Comment (TomShawn) [inline system-variables.md:763]:
```suggestion
- Enabling the read-only mode using this variable only ensures that the entire cluster will finally enter the read-only status. If you have changed the value of this variable in a TiDB cluster but the change has not yet propagated to other TiDB servers, the un-updated TiDB servers are still **not** in the read-only mode.
```

File: system-variables.md [hunk 15]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+- While enabling this variable, the executing SQL statements are not affected. TiDB only checks the read-only status for the SQL statements **to be** executed.
+- While enabling this variable, the uncommitted transactions have the following results:
```
Comment (TomShawn) [inline system-variables.md:765]:
```suggestion
- When this variable is enabled, TiDB handles the uncommitted transactions in the following ways:
```

File: system-variables.md [hunk 16]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+    - For uncommitted read-only transactions, the transactions can be committed normally.
```
Comment (TomShawn) [inline system-variables.md:766]:
```suggestion
    - For uncommitted read-only transactions, you can commit the transactions normally.
```

File: system-variables.md [hunk 17]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+    - For uncommitted transactions that are not read-only transactions, SQL statements executing write operations in these transactions are rejected.
```
Comment (TomShawn) [inline system-variables.md:767]:
```suggestion
    - For uncommitted transactions that are not read-only, SQL statements that perform write operations in these transactions are rejected.
```

File: system-variables.md [hunk 18]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+    - For uncommitted read-only transactions with modified data, commitments of these transactions are rejected.
```
Comment (TomShawn) [inline system-variables.md:768]:
```suggestion
    - For uncommitted read-only transactions with modified data, the commit of these transactions is rejected.
```

File: system-variables.md [hunk 19]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+- After the read-only mode is enabled, all users (including the users with the `SUPER` privilege) cannot execute the SQL statements that might write data unless the user is explicitly granted the `RESTRICTED_REPLICA_WRITER_ADMIN` privileges.
```
Comment (TomShawn) [inline system-variables.md:769]:
```suggestion
- After the read-only mode is enabled, all users (including the users with the `SUPER` privilege) cannot execute the SQL statements that might write data unless the user is explicitly granted the `RESTRICTED_REPLICA_WRITER_ADMIN` privilege.
```

File: system-variables.md [hunk 20]
```diff
@@ -754,6 +754,21 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+- The users with `RESTRICTED_VARIABLES_ADMIN` or `SUPER` privileges can modify this variable. However, if the [Security Enhanced Mode](#tidb_enable_enhanced_security) is enabled, only the users with the `RESTRICTED_VARIABLES_ADMIN` privilege can modify this variable.
```
Comment (TomShawn) [inline system-variables.md:770]:
```suggestion
- Users with `RESTRICTED_VARIABLES_ADMIN` or `SUPER` privileges can modify this variable. However, if the [security enhanced mode](#tidb_enable_enhanced_security) is enabled, only the users with the `RESTRICTED_VARIABLES_ADMIN` privilege can modify this variable.
```

**Comment by en-jin19** [comment]
@TomShawn PTAL, thanks!

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**